pkgs/by-name/op/opcua-commander/package.nix

```diff
@@ -0,0 +1,40 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, typescript
+, esbuild
+, makeWrapper
+, nodejs
+}:
+buildNpmPackage rec {
+  pname = "opcua-commander";
+  version = "0.37.0";
+
+  src = fetchFromGitHub {
+    owner = "node-opcua";
+    repo = "opcua-commander";
+    rev = version;
+    hash = "sha256-wQXSSNinY85Ti+D/zklYP2N8IP3OsN9xQNJuuQr4kVU=";
+  };
+
+  npmDepsHash = "sha256-Ux1X/3sam9WHrTfqoWv1r9p3pJOs6BaeFsxHizAvjXA=";
+  nativeBuildInputs = [ esbuild typescript makeWrapper ];
+
+  postPatch = ''
+    substituteInPlace package.json \
+      --replace-warn "npx -y esbuild" "esbuild"
+  '';
+
+  postFixup = ''
+    wrapProgram $out/bin/opcua-commander \
+      --prefix PATH : "${lib.makeBinPath [nodejs]}"
+  '';
```

Comment:
```suggestion
```
It's already wrapped:
```
cat result/bin/opcua-commander
#! /nix/store/5lr5n3qa4day8l1ivbwlcby2nknczqkq-bash-5.2p26/bin/bash -e
exec "/nix/store/rlv9fcq70pniyrm3xy21wgk5rdhdh12h-nodejs-20.11.1/bin/node"  /nix/store/3z6nn7z4q3l15fm8y5jrcnjcscgi5hk6-opcua-commander-0.37.0/lib/node_modules/opcua-commander/bin/opcua-commander "$@"
```

Reply:
yeah, but if `nodejs` is not in PATH, the application fails to connect to any OPC-UA server with:
```
using nodejs crypto (native)
Some package version(s) failed!
TypeError: Cannot read properties of undefined (reading 'version')
    at check_nodejs (/nix/store/2q4wxc9zh580yxmxjjw7vciqh3v2kbkw-opcua-commander-0.37.0/lib/node_modules/opcua-commander/bin/bundle.js:97497:75)
    at /nix/store/2q4wxc9zh580yxmxjjw7vciqh3v2kbkw-opcua-commander-0.37.0/lib/node_modules/opcua-commander/bin/bundle.js:97524:3
```
this only happens when trying to connect to a server with the endpoint flag (the --help does work however) for example:
`./result/bin/opcua-commander -e opc.tcp://192.168.0.5:4840`

Reply:
I see

Reply:
Maybe it worth to add some comment to postFixup, otherwise someone can drop it in the future without proper testing with real OPC-UA server.

Reply:
I've added a comment with instructions to run the app against the demo server of the organization that created the utility. That should be available for anyone, no need to setup your own server for testing.